fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java

```diff
@@ -884,8 +886,11 @@ public void analyze(TQueryOptions tQueryOptions) throws UserException {
                 || parsedStmt instanceof InsertStmt
                 || parsedStmt instanceof CreateTableAsSelectStmt) {
             if (Config.enable_resource_group && context.sessionVariable.enablePipelineEngine()) {
-                analyzer.setResourceGroups(analyzer.getEnv().getResourceGroupMgr()
-                        .getResourceGroup(context.sessionVariable.resourceGroup));
+                ResourceGroup resourceGroup = analyzer.getEnv().getResourceGroupMgr().getResourceGroup(
+                        context.sessionVariable.resourceGroup);
+                List<TPipelineResourceGroup> resourceGroups = Lists.newArrayList();
+                resourceGroups.add(resourceGroup.toThrift());
```

Comment:
How to ensure thread safety of resourceGroup when multiple threads are executing toThrift and alterResourceGroup?

Reply:
So we should write another func named with `getResourceGroupThrift` to get return value with type of ` List<TPipelineResourceGroup>`. What do you think?